com.microsoft.mrtk.spatialmanipulation/Solvers/TapToPlace.cs

```diff
@@ -489,24 +518,40 @@ private void RegisterPickupAction()
             interactable.OnClicked.AddListener(StartPlacement);
         }
 
+        /// <summary>
+        /// Registers the event which performs placement.
+        /// </summary>
+        private void RegisterPlacementEvent()
+        {
+            if (interactable == null)
+            {
+                Debug.Log("Failed to register the placement event. There is no StatefulInteractable set.");
+                return;
+            }
+            interactable.OnClicked.AddListener(StopPlacement);
+        }
+
         /// <summary>
         /// Unregisters the input action which performs placement.
         /// </summary>
         private void UnregisterPlacementAction()
         {
-            InputAction placementAction = GetInputActionFromReference(placementActionReference);
-            if (placementAction == null)
+            foreach (InputAction placementAction in placementActions)
             {
-                Debug.Log("Failed to unregister the placement action, the action reference was null or contained no action.");
-                return;
+                if (placementAction == null)
+                {
+                    Debug.Log("Failed to unregister the placement action, the action reference was null or contained no action.");
+                    return;
+                }
+                placementAction.performed -= StopPlacementAction;
             }
-            placementAction.performed -= StopPlacement;
+            lastActions = new List<InputAction>();
         }
 
         /// <summary>
         /// Unregisters the event which performs pickup.
         /// </summary>
-        private void UnregisterPickupAction()
+        private void UnregisterPickupEvent()
         {
             if (interactable == null)
```

Comment:
since interactable is a required component, a null check doesn't make sense. This should be exceptional and cause an exception.